### PR TITLE
router: PathPrefix router was wrong.

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Current router path handling was wrong, it was working fine due to some mutability inside gorilla/mux but this is an important bug. 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by vadmeste** [comment]
Can we discuss this before you merge? there is a story behind this.

**Comment by harshavardhana** [comment]
> Can we discuss this before you merge? there is a story behind this.

Just pinged you what is the story?

**Comment by fwessels** [comment]
LGTM

**Review by krishnasrinivas — APPROVED** [review]

File: cmd/api-router.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ func registerAPIRouter(mux *router.Router) {
 	}
 
 	// API Router
-	apiRouter := router.NewRouter().PathPrefix("/").Subrouter()
+	apiRouter := mux.NewRoute().PathPrefix("/").Subrouter()
```
Comment (vadmeste) [inline cmd/api-router.go]: This has bad effect on the handlers per route below (time validity handler), but looks like it works like this:

```
apiRouter := mux.NewRoute().PathPrefix("").Subrouter()
```

Isn't this more correct ? especially '/' is already added in bucket router.

Comment (harshavardhana) [reply cmd/api-router.go]: I dont see any tests failing ?

Comment (harshavardhana) [reply cmd/api-router.go]: Also ping me on how to reproduce it will see a cleaner fix. The existing route is wrong and can lead to bugs.

**Review by vadmeste — APPROVED** [review]

**Pull request merged**